### PR TITLE
Fix detection of renaming files when the client is stopped

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: src/local/watcher.js [hunk 1]
```diff
@@ -233,7 +233,9 @@ class LocalWatcher {
               log.info({path: filePath}, 'file added')
               this.prep.addFileAsync(SIDE, doc).catch(logError)
             } else {
-              const same = find(docs, d => this.pending.hasPath(d.path))
+              const same = find(docs, this.paths
+                  ? d => !fs.existsSync(d.path)
+                  : d => this.pending.hasPath(d.path))
```
Comment (sebn) [inline src/local/watcher.js]: I believe we should at some point either rename `this.path` to `this.initialPaths` and have `this.isInitialScan = () => this.initialPaths != null`, or even have some kind of `this.initialScan = {paths: []}` object (so we can do `if (this.initialScan) this.initialScan.paths.push(...)` and we don't have a remaining useless `this.initialPaths` property once scan is completed).

Comment (sebn) [inline src/local/watcher.js]: (can be done later thought, just wanted to share my thoughts)

File: src/local/watcher.js [hunk 2]
```diff
@@ -338,13 +340,14 @@ class LocalWatcher {
...
+              this.onUnlinkFile(doc.path)
             } else {
-              log.info({path: doc.path}, 'deleted while client was stopped')
-              await this.prep.trashDocAsync(SIDE, doc)
+              this.onUnlinkDir(doc.path)
             }
           }
           delete this.paths
-          callback()
+          setTimeout(callback, 3000)
```
Comment (sebn) [inline src/local/watcher.js]: I assume having the usual unlinked file/dir timeouts + waiting 3 seconds after initial scan prevents the "onready" race condition?

**Comment by sebn** [comment]
Arg... PR rebase & merge takes the last commit message instead of the PR one 😭

**Comment by sebn** [comment]
My bad, it actually did a rebase as usual. Should have squashed instead.

File: src/local/watcher.js (hunk 2 above)
Comment (nono) [inline src/local/watcher.js]: Exactly

**Pull request merged**